### PR TITLE
Unreleased changelog folder

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation infrastructure.

- [x] Corresponding documentation was updated

~~I'm submitting this as a WIP to demonstrate what the future of the unreleased changelog will be. This should be coupled with a script extracting the relevant changelog entries for a new release~~ (à la https://about.gitlab.com/2018/07/03/solving-gitlabs-changelog-conflict-crisis/), whose two advantages will be:
- no more CHANGES conflict
- no more mistake about which CHANGES section to put the entry in, especially when the PR is proposed for backporting but might be rejected, or if the PR has been waiting for a long time and the location has stalled.

The main feature demonstrated by this PR and which goes beyond what GitLab proposed is that unreleased changes are still visible in a central location (the deployed development version of the reference manual).

TODO

- [x] Remove commit with fake entries for demonstration purpose.
- [x] Wait for legacy Makefile to be removed, or update `Makefile.doc` as well.
- [x] Wait for #9939 to be merged first.

TODO for future PR

- [ ] Create the script for extracting the changelog for a new release.

### CODE REVIEW

**Comment by ejgallego** [comment]
That's cool, would be even cooler if Gitlab/Github could maintain the files for us and understand the PR numbers in them.

**Comment by Zimmi48** [comment]
What do you have in mind?

**Comment by ejgallego** [comment]
> What do you have in mind?

Imagine that gitlab would know all the PRs in master but not on 8.10, and these PRs have the tag `changes`, then it could generate the folder with the commit messages automaticaly..

I am just brainstorming tho.

**Comment by Zimmi48** [comment]
This kind of brainstorming is quite welcome. It could be seen as a next step if we can determine exactly what we want. This could be implemented with the bot, for instance.

**Comment by Zimmi48** [comment]
Here's what it gives on these fake changelog entries: https://coq.gitlab.io/-/coq/-/jobs/197752644/artifacts/_build/default/doc/sphinx_build/html/changes.html

File: doc/changelog/README.md
Comment (gares) [inline doc/changelog/README.md]: Isn't this redundant given that the file name is already containing NNNNN?

File: doc/changelog/README.md
Comment (SkySkimmer) [inline doc/changelog/README.md]: This looks pretty heavy compared to our current format.

File: doc/changelog/README.md
Comment (Zimmi48) [reply doc/changelog/README.md]: Yes, it is, and at some point we might switch to a more clever compilation script that will add such annotations by itself, but for now, it is simpler like this.

File: doc/changelog/README.md
Comment (Zimmi48) [reply doc/changelog/README.md]: Reduced the heaviness by reusing a format closer to the one from Dune changelog, but with full names. Already applied in the credits PR.

**Comment by Zimmi48** [comment]
We are going to need to merge this soon and I wouldn't mind a hand to port the build changes to the Makefile (since obviously we are not going to be able to wait for #8729).

**Comment by gares** [comment]
I was thinking that with this approach we lose the organization of changes into sections.
Is it doable to have a subdir per section?

**Comment by gares** [comment]
>  I wouldn't mind a hand to port the build changes to the Makefile

We need to do that only if dune misses also 8.11 right?

**Comment by Zimmi48** [comment]
> I was thinking that with this approach we lose the organization of changes into sections.
> Is it doable to have a subdir per section?

Yes, it should be doable but then I think we should decide section names before-hand (as opposed to letting PRs authors invent them, which often leads to duplication and inconsistencies).

> > I wouldn't mind a hand to port the build changes to the Makefile
> 
> We need to do that only if dune misses also 8.11 right?

No, because this PR should be merged as early as possible, whereas the removal of legacy Makefiles can still wait several weeks or months if needed.

**Comment by Zimmi48** [comment]
@gares I pushed a new commit in which there is such a division in categories. Here is the rendered result: https://coq.gitlab.io/-/coq/-/jobs/204895434/artifacts/_build/default/doc/sphinx_build/html/changes.html

**Comment by gares** [comment]
Awesome! do you want me to fix the Makefile?

**Comment by Zimmi48** [comment]
Yes, that would be very appreciated. Feel free to push here. I can take care of the rebase later.

**Comment by Zimmi48** [comment]
ping @gares

**Comment by gares** [comment]
Sorry I did not manage last week, I'll do my best on Monday

**Comment by gares** [comment]
I be pushed a commit but it needs to be released in order to be tested. I hope I got the paths right.

**Comment by Zimmi48** [comment]
It turns out that the two entries moved from `CHANGES.md` to `doc/changelog` were backported already, so it is fine to backport this PR as well. A subsequent PR will ensure that these entries are integrated in the 8.10+beta1 changes before the beta release.

**Comment by Zimmi48** [comment]
Would appreciate getting an assignee and this PR merged before another conflict arises.

**Comment by Zimmi48** [comment]
Too late. New conflict.

**Comment by Zimmi48** [comment]
@vbgl After this new rebase, the current version of the PR cannot be backported as-is: the changelog entry `doc/changelog/04-tactics/10059-change-no-check.rst` would have to be dropped, unless the four PRs that are mentioned in it (#10012, #10017, #10053, #10059) are backported too.

**Comment by vbgl** [comment]
Thanks Théo for the hint. I’ll follow your advice for back-porting after this one is merged.

**Comment by gares** [comment]
OK, I'll do it.

**Review by gares — APPROVED** [review]

**Comment by Zimmi48** [comment]
@vbgl I think that it would make sense for you to be this PR's assignee, as the 8.10 RM. Note that even if there are quite a few code owners, all the changes to files that they own are quite trivial, so I wouldn't wait for everyone's approval.

**Review by vbgl — APPROVED** [review]

**Pull request merged**
